### ISSUE TITLE
Remove links pasting

### ISSUE THREAD

**Issue by Andrea-28-03** [kxxt/obsidian-advanced-paste#15]
Hi,
I use [perplexity](https://www.perplexity.ai/) and it makes citations that when i normally copy and paste them on obsidian they breake and make a lot of blanks lines,
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/8c09721a-3a3e-4599-a97f-60dcb4db4803)
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/5989ce8a-98db-4525-9031-d07e1847abb3)
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/ee19bab8-dc5e-4462-928b-2f26d8dddbcb)
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/33bdeb73-f6b1-4518-bedc-c67d0b70b0e9)
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/f559a5c6-faa4-4d7c-8129-ce1a1ae49503)

Smart join is incredible but it delete all the links and tabulations.
Is there a way to fix it also with an custom trasform it would be really usefull 
It does this only with that site and can't understand why...

**Comment by kxxt** [comment]
> Smart join is incredible but it delete all the links and tabulations.

Because smart join operates in pure text mode. And it is (almost) intended for pasting from PDF.

> Is there a way to fix it also with an custom trasform it would be really usefull

Yes, a custom transform could do it.

From the picture you provided, It seems that there are spaces and newlines inside markdown links. I checked it and it looks like a bug in turndown.

So if you want to write a custom transform to do the task, note that remark will produce a bad syntax tree because of it. Of course you can also use regex instead of remark to avoid it.

**Comment by kxxt** [comment]
The following script might helps. It uses a simple regex.

```js
export async function stripWhiteSpacesInLinks(input, { turndown }) {
  if (input.types.includes("text/html")) {
    const html = await input.getType("text/html");
    const md = turndown.turndown(await html.text());
    return md.replace(/\[\n*(.*)\n*\]\((.*)\)/gm, "[$1]($2)");
  }
  return { kind: "err", value: "No html found in clipboard!" };
}
stripWhiteSpacesInLinks.type = "blob";
```

**Comment by Andrea-28-03** [comment]
No, I tried it, it works exactly like the normal copy and paste. I think it can't find the link...
![image](https://github.com/kxxt/obsidian-advanced-paste/assets/62112468/7e6f8037-8a05-4a51-8a04-844df3135966)
I tried some different replacement or founding methods, but them didn't change anything 

```.js
export async function fanculo(input, { turndown }) {
  if (input.types.includes("text/html")) {
    const html = await input.getType("text/html");
    const md = turndown.turndown(await html.text());
    return md.replace(/\[\n*(.*)\n*\]\(html:\\(.*)\)/sgu, " [$1]($2)");
    /*return md.replace(/\[.*(\d+).*\].*?\(html:\\(.*?)\)/gms, function (match, text, url) {
      // Personalizza la formattazione dei link come desiderato
      return ` [${text}](${url})`;*/
    }

  return {kind: "err", value: "No html found in clipboard!"};
}
fanculo.type = "blob";
```

**Comment by kxxt** [comment]
The script works for me.

![Peek 2023-11-05 19-28](https://github.com/kxxt/obsidian-advanced-paste/assets/18085551/7130d675-9894-4e40-b05f-a32000d17ac3)

It seems I can't expect this bug to be fixed on turndown's side: https://github.com/mixmark-io/turndown/pull/419 

When I get more time I will fix it in the default paste command, maybe by maintaining a fork of turndown.

**Comment by Andrea-28-03** [comment]
You are right, using [perplexity](https://www.perplexity.ai/) from the official website and then pasting it with your script does work, really great.
But from the [custom frame](https://github.com/Ellpeck/ObsidianCustomFrames) of perplexity it doesn't, it would be great if you knew a workaround about that, but it is ok also like that, thanks
